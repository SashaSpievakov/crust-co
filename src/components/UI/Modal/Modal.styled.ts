import styled, { css } from 'styled-components';
import { BsXLg } from 'react-icons/bs';

import { ButtonPay, ButtonBack } from 'src/styles/Buttons.styled';

const mixinArticle = css`
  position: fixed;
  top: 40%;
  left: 50%;
  text-align: center;
  background: ${({ theme }) => theme.colors.fourth};
  transform: translate(-50%, -40%);
  z-index: 10;
  border-radius: 15px;
  box-shadow: 0 5px 20px 0 rgba(0, 0, 0, 0.05);

  @media screen and (max-width: 500px) {
    top: 35%;
    transform: translate(-50%, -35%);
  }
`;

const Bg = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
  width: 100vw;
  height: 100vh;
  z-index: 0;
  background-color: rgba(0, 0, 0, 0.3);
  transform: translate(-50%, -50%);
`;

const Aricle = styled.article`
  ${mixinArticle}
  width: 40%;
  height: 95%;

  @media screen and (max-width: 900px) {
    width: 70%;
    height: 95%;
  }

  @media screen and (max-width: 500px) {
    width: 80%;
    height: 90%;
  }
`;

const AricleSubmitted = styled.article`
  ${mixinArticle}
  width: 40%;
  height: 30%;

  @media screen and (max-width: 900px) {
    width: 70%;
    height: 40%;
  }

  @media screen and (max-width: 500px) {
    width: 80%;
  }
`;

const Header = styled.h3`
  font-size: 2.5rem;
  margin-top: 5%;

  @media screen and (max-width: 700px) {
    font-size: 2rem;
  }

  @media screen and (max-width: 500px) {
    font-size: 1.7rem;
  }
`;

const SubHeader = styled.h4`
  font-size: 1.3rem;
  font-weight: 400;
  width: 70%;
  margin: 0 auto 3%;

  @media screen and (max-width: 700px) {
    font-size: 1rem;
  }

  @media screen and (max-width: 500px) {
    font-size: 0.75rem;
  }
`;

const Cross = styled(BsXLg)`
  position: absolute;
  top: 15px;
  right: 15px;
  font-size: 25px;
  opacity: 0.3;
  border-radius: 5px;
  cursor: pointer;

  &:hover {
    opacity: 0.7;
  }

  &:focus {
    outline: 3px solid ${({ theme }) => theme.colors.third};
    outline-offset: 3px;
  }
`;

const SubHeaderSubmitted = styled.h4`
  font-size: 1.3rem;
  font-weight: 400;
  width: 60%;
  margin: 2% auto 0;

  @media screen and (max-width: 700px) {
    font-size: 1rem;
  }
`;

const Form = styled.form`
  display: flex;
  flex-direction: column;
  align-items: center;
  row-gap: 2em;
  width: 60%;
  margin: 0 auto;

  @media screen and (max-width: 900px) {
    row-gap: 1.5em;
  }

  @media screen and (max-width: 500px) {
    width: 75%;
  }
`;

const Group = styled.div`
  display: flex;
  width: 100%;
  flex-direction: column;
  align-items: start;
`;

const Label = styled.label`
  font-size: 1.3em;
  font-weight: 700;

  @media screen and (max-width: 700px) {
    font-size: 1rem;
  }
`;

const Input = styled.input`
  font-size: 1.2em;
  width: 100%;
  padding: 0.5em 0.7em;
  color: ${({ theme }) => theme.colors.third};
  background-color: ${({ theme }) => theme.colors.gray};
  border: 3px solid ${({ theme }) => theme.colors.secondaryTransparent};
  border-radius: 10px;

  &:focus {
    border: 3px solid ${({ theme }) => theme.colors.secondary};
  }

  @media screen and (max-width: 700px) {
    padding: 0.3em 0.5em;
  }
`;

const ModalButton = styled(ButtonPay)`
  margin-top: 5%;
`;

const SubmittedButton = styled(ButtonBack)`
  margin-top: 5%;
`;

export {
  Bg,
  Aricle,
  AricleSubmitted,
  Header,
  SubHeader,
  Cross,
  SubHeaderSubmitted,
  Form,
  Group,
  Label,
  Input,
  ModalButton,
  SubmittedButton,
};
