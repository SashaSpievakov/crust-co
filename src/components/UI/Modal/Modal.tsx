import { Bg, Aricle } from './Modal.styled';

interface ModalProps {
  setIsOpen: (value: boolean) => void;
}

const Modal = ({ setIsOpen }: ModalProps) => {
  return (
    <Bg onClick={() => setIsOpen(false)}>
      <Aricle>some texxt</Aricle>
    </Bg>
  );
};
export default Modal;
