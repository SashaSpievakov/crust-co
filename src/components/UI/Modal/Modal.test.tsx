import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Modal from './Modal';
import rendererWithAllProviders from '../../../tests/helpers/rendererWithProviders';
import renderWithProviders from '../../../tests/helpers/renderWithProviders';

describe('Modal Tests', () => {
  test('renders the Modal UI component', () => {
    const snapshot = rendererWithAllProviders(<Modal setIsOpen={jest.fn()} />);
    expect(snapshot).toMatchSnapshot();
  });

  describe('checks select logic', () => {
    test('checks thin type click', async () => {
      const price = await screen.findByRole('heading', {
        level: 3,
      });
      const typeThin = screen.getByText(/thin/i);

      userEvent.click(typeThin);

      expect(price).toHaveTextContent(/15\$/i);
    });
  });
});
