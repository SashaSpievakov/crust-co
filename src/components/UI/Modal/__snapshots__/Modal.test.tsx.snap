// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Tests renders the Modal UI component 1`] = `
Object {
  "0": <div
    className="sc-ftTHYK eGxASH"
    onClick={[Function]}
  />,
  "1": <article
    className="sc-pyfCe cMErXB"
  >
    <h3
      className="sc-kDvujY gpEeWv"
    >
      Order
    </h3>
    <h4
      className="sc-ipEyDJ gdDGzU"
    >
      Please provide your information, and our operator will get in touch with you as soon as possible to finish your order
    </h4>
    <svg
      className="sc-csuSiG kBcuim"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      onKeyDown={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      tabIndex={0}
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1.293 1.293a1 1 0 0 1 1.414 0L8 6.586l5.293-5.293a1 1 0 1 1 1.414 1.414L9.414 8l5.293 5.293a1 1 0 0 1-1.414 1.414L8 9.414l-5.293 5.293a1 1 0 0 1-1.414-1.414L6.586 8 1.293 2.707a1 1 0 0 1 0-1.414z"
      />
    </svg>
    <form
      className="sc-bqWxrE gaAleb"
      onSubmit={[Function]}
    >
      <div
        className="sc-ksBlkl ihNhWJ"
      >
        <label
          className="sc-hBxehG gZLzlN"
          htmlFor="inputName"
        >
          Name
        </label>
        <input
          className="sc-fnGiBr ktBbpq"
          id="inputName"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
        />
      </div>
      <div
        className="sc-ksBlkl ihNhWJ"
      >
        <label
          className="sc-hBxehG gZLzlN"
          htmlFor="inputPhone"
        >
          Phone Number
        </label>
        <input
          className="sc-fnGiBr ktBbpq"
          id="inputPhone"
          name="phone"
          onBlur={[Function]}
          onChange={[Function]}
          onKeyPress={[Function]}
          type="number"
        />
      </div>
      <div
        className="sc-ksBlkl ihNhWJ"
      >
        <label
          className="sc-hBxehG gZLzlN"
          htmlFor="inputCity"
        >
          City
        </label>
        <input
          className="sc-fnGiBr ktBbpq"
          id="inputCity"
          name="city"
          onBlur={[Function]}
          onChange={[Function]}
        />
      </div>
      <div
        className="sc-ksBlkl ihNhWJ"
      >
        <label
          className="sc-hBxehG gZLzlN"
          htmlFor="inputAdddress"
        >
          Adddress (optional)
        </label>
        <input
          className="sc-fnGiBr ktBbpq"
          id="inputAdddress"
          name="address"
          onBlur={[Function]}
          onChange={[Function]}
        />
      </div>
      <button
        className="sc-bcXHqe sc-hLBbgP sc-fEXmlR dRXTCo czoOAi mJgNn"
        type="submit"
      >
        <span>
          Send
        </span>
      </button>
    </form>
  </article>,
}
`;
